### DISCUSSION TITLE
v1.2.0 Discussion

### DISCUSSION THREAD

**Discussion by mattroumaya** [Gilead-Public/gsm#628] [General]
Two primary use cases for `{gsm}`:
**1. Facilitate automated nightly data cuts for Gizmo**
  - THE BLOB - we're not going to send them blobs, will need to parse into a series of flat files.
    - Structure of flat files needs detailed spec
    - For each study each night, the final transfer needs to be a set of `.csv` files from a few different data sources
    - Across all data sources, something like ~50 `.csv` files per study
    - Over half will be Gizmo outputs, and the rest will be metadata that inform how the data model is run and provides metadata about the model, how to interpret analytics/KRI, etc.
    - ZS source-to-target mapping is mostly done - we need to make sure we're in agreement, providing expected outputs, and document the specs.
    - Avoid redundancy, be explicit about mapping. 
    - KRILabel and ScoreLabel - probably not necessary to provide as a column. 
  - Need to consider where all metadata lives. 
    - KRI spreadsheet should probably be saved directly in `{gsm}` that gets updated with each release
    - KRIID, QTLID, need to iron out metadata specs
    - Might get rid of "label" attribute in `Study_Assess()` and instead tie to a metadata structure to be defined.

**2. Let analysts directly query the data**
  - This is mostly good to go, is well documented in vignettes. 
  - Figure out best way to connect Gizmo/code export. We should be able to identify code that was initially run in Gizmo so that it's entirely reproducible.
  - Less automated - analyst notices something in Gizmo and is able to drill-down and look at specific/granular data
  - Maybe re-run on subset of data, additional filters, strata, etc.

**Qualification:**
1. Shouldn't let this blow up qualification for `v1.1.0`. In general, we're moving away from a blob - this structure will be changing a lot. 
2. Long term - have qualification focus on the specific pieces that are relevant to the use case. No need to focus on the entire workflow/output while a lot of it is TBD.
3. Fine to skip over/hack some of the qualification pieces for now, until metadata structure is more locked down.

**Comment by jwildfire** [comment]
Epics/Issues
- Data Pipeline Updates - Update clindata to match defaults in Raw+ data model
    - #611 
- Workflow Updates - Standardize workflow process 
    - Define and pass through some kind of WorkflowID 
    - Make it easy to ingest settings (thresholds, KRIs, etc) for a given study and run the requested workflows. 
- Data Export - Create flat files capturing study-level results for a given data cut. 
   - Create detailed spec for all needed exports as Appendix 2 in the Data Pipeline vignette
   - Consider whether we should capture the study-level raw+ `mapping` as an object in the data model. 
   - #618 
   - #619 
   - #621

**Comment by mattroumaya** [comment]
Notes for v1.1.0 Recap: 

1. In Gizmo v1, need to run by country and that's it. Hardcode `CountryID` instead of `CustomGroupID`.
- We'll have SiteID, StudyID, and CountryID in Gizmo by default, fine to include `CustomGroupID` as well 
- Won't calculate region KRIs, but might filter for a specific region
- Only place we're currently surfacing Country KRI is on the longitudinal charts
- **TODO** -- file issue for CountryID (v1.2.0)

2. **TODO**: File issue for v1.2/1.3 to test stratified assessments/updated `MakeStratifiedAssessment()` before implementation for `{gizmo}` MVP/v1 
- Update `Visualize_Scatter()` to work with everything - showing numerator/denominator
- Everything also gets a bar chart for KRI and Score

3. Whenever GroupID is StudyID, we'll run `analyze_Identity()` by default (for QTLs)

4. TODO: file issue for sanity check on `Disp_Assess()`
- Update parameterization for `Disp_Assess()`
- Look at `Transform_EventCount()` to make sure we're capturing the denominator correctly (total # of people)
  - Total N should be total # of people
- KRI will be Count / Total

5. For KRI - remove red threshold line, color bars by whether they're flagged or not
- File issue for v1.2 - create table for `Visualize_Score()`
- One row per site - x-axis is the KRI/Score value

6. **TODO**: File issue to remove barchart for `Visualize_Count()`

7. **TODO**: Add `Analyze_Identity()` to labs
